src/components/Company/Offers/Manage/CompanyOffersManagementWidget.spec.js

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { screen } from "@testing-library/react";
+
+import CompanyOffersManagementWidget from "./CompanyOffersManagementWidget";
+import { renderWithStore } from "../../../../test-utils";
+import { MuiPickersUtilsProvider } from "@material-ui/pickers";
+import DateFnsUtils from "@date-io/date-fns";
+
+
+describe("App", () => {
+    test("Renders Component", async () => {
+        renderWithStore(
+            <MuiPickersUtilsProvider utils={DateFnsUtils}>
+                <CompanyOffersManagementWidget />
+            </MuiPickersUtilsProvider>
+        );
+
+        // Use getBy by default when element should be available
+        expect(screen.getByText("Loading...")).toBeInTheDocument();
+
+        expect(screen.getByRole("heading")).toBeInTheDocument();
+
+        // Use queryBy When asserting for missing element
+        expect(screen.queryByText("Offers Management")).toBeNull();
+
+        screen.debug();
+
+        // Use find by for elements that aren't there initially, but will be eventually
+        expect(await screen.findByText("Offers Management")).toBeInTheDocument();
```

Comment:
I believe that if you use act() on the render above, you won't need the await find* query, please try it with a normal get and see if it works, as it will be faster

Reply:
I've tried to use only with the act and it was not working. From what I've searched, since render() is a synchronous function, it only flushes synchronous state updates and that was not the case.

Reply:
render is synchronous, but act is not :P You can await act(()=>{render...}), and from what I know, it will execute all the async code at once ( the batch changes from useEffects, setStates etc as if they were running in a browser (where they happen in batches, not synchronously)